src/herder/TxSetFrame.cpp

```diff
@@ -230,25 +233,55 @@ TxSetFrame::trimInvalid(Application& app,
             int64_t newBalance =
                 lastTx->getSourceAccount().getBalance() - totFee;
             if (newBalance < lastTx->getSourceAccount().getMinimumBalance(
-                                 app.getLedgerManager()))
+                  app.getLedgerManager()) && !processLastInvalidTxLambda(item.second, lastTx))
```

Comment:
I think `processLastInvalidTxLambda` should be:
* renamed to `processInsufficientBalance`
* move the call inside the block (right before returning `false`), as Trim doesn't care about the return value
* ditch the second parameter (lastTx) as it's I think equal to item.second.back()